[FEATURE] Add plugin to allow oidc authentication without login box

SomeBdyElse opened the pull request.

xperseguers commented on README.md:
```diff
@@ -8,6 +8,25 @@ If you are a Swiss Alpine Club section, be sure to get in touch with Bern in ord
 
+## Default FE Loginbox
+
+This extension integrations with the felogin systemextension and provides a new marker ###OPENID_CONNECT### to be used 
```
- integrations => integrates
- systemextension => system extension

xperseguers commented on README.md:
```diff
@@ -8,6 +8,25 @@ If you are a Swiss Alpine Club section, be sure to get in touch with Bern in ord
+## Default FE Loginbox
+
+This extension integrations with the felogin systemextension and provides a new marker ###OPENID_CONNECT### to be used 
+in the felogin template. A sample template is included. Once the users clicks the new openid connect link, he/she will 
```
- the users clicks => the user clicks
- he/she => she (I figured out by reading many presentations/texts from native speakers that it's very common to use the feminine form when speaking about a user)

xperseguers commented on README.md:
```diff
@@ -8,6 +8,25 @@ If you are a Swiss Alpine Club section, be sure to get in touch with Bern in ord
+
+This extension integrations with the felogin systemextension and provides a new marker ###OPENID_CONNECT### to be used 
+in the felogin template. A sample template is included. Once the users clicks the new openid connect link, he/she will 
+redirected to the ids server. 
```
- ids => IDS

xperseguers commented on README.md:
```diff
@@ -8,6 +8,25 @@ If you are a Swiss Alpine Club section, be sure to get in touch with Bern in ord
+
+## OIDC Login
+
+If openid_connect is your only means of fe login, you can use the included "OIDC Login" plugin. Add it to your login 
```
- fe login => either FE login or better Frontend login

xperseguers commented on Classes/Controller/LoginController.php:
```diff
@@ -0,0 +1,108 @@
+<?php
+
```
Missing standard TYPO3 header block

xperseguers commented on Classes/Controller/LoginController.php:
```diff
@@ -0,0 +1,108 @@
+    /**
+     * @var ContentObjectRenderer
+     *
+     * Will be injected automatically, if this controller is called as a plugin
```
Comment should go before the `@var` declaration

xperseguers commented on Classes/Controller/LoginController.php:
```diff
@@ -0,0 +1,108 @@
+     * @param string $_ ignored
+     * @param array|null $pluginConfiguration
+     *
+     * Main entry point for the OIDC plugin.
```
comment goes *before* the parameters

xperseguers commented on Classes/Controller/LoginController.php:
```diff
@@ -0,0 +1,108 @@
+     * If the user has just been logged in and just came back from the authorization server, redirect the user to the
+     * final redirect URL.
+     */
+    public function login($_ = '', ?array $pluginConfiguration)
```
`?array` is not compatible with PHP 5.5 which is still a dependency defined in `composer.json`

Note that I'd be fine to raise the dependency to PHP 7.2, and drop TYPO3 7.6 support as well but this needs to be done separately.

xperseguers commented on Classes/Controller/LoginController.php:
```diff
@@ -0,0 +1,108 @@
+
+        if (GeneralUtility::_GP('logintype') == 'login') {
+            $this->performRedirectAfterLogin();
+            // performRedirectAfterLogin stops flow by emitting a redirect
```
I'd prefer the comment before the call

xperseguers commented on Classes/Controller/LoginController.php:
```diff
@@ -0,0 +1,108 @@
+
+        $state = $service->getState();
+        $_SESSION['oidc_state'] = $state;
+        $_SESSION['oidc_login_url'] = $_SERVER['REQUEST_URI'];
```
Any reason for not using the official TYPO3 API `GeneralUtility::getIndpEnv()`?

SomeBdyElse commented:
@xperseguers Thank you for the valuable comments. I addressed them by: 
* Updating the README, correcting my spelling
* Fixed comments in the new controller class
* Changed typing to remain compatible with the older PHP version
* Removed an unused import (unrelated)

SomeBdyElse commented:
My last commit is a "fixup". I suggest to **squash**-merge the feature branch into master.
